### ISSUE TITLE
A way to work with alpha video（WPF）

### ISSUE THREAD

**Issue by LazyCuteLion** [SuRGeoNix/Flyleaf#492]
1、FlyleafHost line-880,change the border.background to #01000000
2、Renderer.PixelShader line-560 cancel this line,does not use BlendState（if you did not do it,then you will got repetitive frame of the video）
3、Renderer.Present line-126 and 247 cancel these two line, does not use ClearRenderTargetView,beacuse of it will ignore the A of the second parameters(even it is Color4) and give a solid color.
4、there have a apng to test,you also can converter to a mov
![qie_640x360](https://github.com/user-attachments/assets/1de4bcad-201d-4dc0-bf35-382d0aef5e1b)
5、this has some flaws,there are burrs on the edges (same video,[ffmediaelement](https://github.com/unosquare/ffmediaelement) rendering good). hope someone can fix it.
![Snipaste_2024-08-02_16-40-06](https://github.com/user-attachments/assets/4269dcc9-2fe7-48fc-80b8-1de0dd4eb88d)

**Comment by LazyCuteLion** [comment]
on the last point, some progress has been made:
ShaderCompiler line-366 add "color.rgb *= color.a;"

**Comment by qinjianzhanqing** [comment]
> on the last point, some progress has been made: ShaderCompiler line-366 add "color.rgb *= color.a;"

Hello, thank you very much for providing the method. I have tested it, but there is a problem. It is that playing only one transparent channel video is not a problem. However, when I switch to a video with a transparent channel, the window interface freezes and the second video cannot be played normally.

**Comment by LazyCuteLion** [comment]
https://github.com/LazyCuteLion/Flyleaf/commit/86596a7ae91ae270019b90f91418c0c45bf8a3b2

**Comment by LazyCuteLion** [comment]
https://github.com/LazyCuteLion/Flyleaf/tree/alpha_video

**Comment by LazyCuteLion** [comment]
WinUI3 has some progress: use [WinUIEx](https://dotmorten.github.io/WinUIEx/concepts/CustomBackdrops.html#transparent-backdrop),set the Window.SystemBackdrop to TransparentTintBackdrop, and change the line-157 of  Renderer.SwapChain.cs(set alpha to true of GetSwapChainDesc).Then,it renderering as same of WPF,but,it can't be mouse through.
I think,the FlyleafHost(WinUI3) should be change as same of WPF,use a Window host the SwapChainPanel.
Do you have any time to try it? @SuRGeoNix

**Comment by SuRGeoNix** [comment]
Not really, maybe in 1-2 weeks... I'm trying to release v3.8 with updated vortice/custom generator-ffmpeg libraries and some other libs that will be the entry level for v4.

**Comment by LazyCuteLion** [comment]
Ok.But I really want it can support alpha.
Look forward to your new version.

**Comment by LazyCuteLion** [comment]
`
Surface.Background = new SolidColorBrush(new Color() { A = 1});
Config.Video.BackgroundColor = new(0,0,0,0);
`
I try it,but it does not my want.

for example:

` 
<Grid>
     <Image Stretch="Fill" Source="bg.jpg" Name="bg"/>
     <fl:FlyleafHost Player="{Binding Player}" ……
`

i need to can see the "bg",but not show some color of fill with the video view.

**Comment by SuRGeoNix** [comment]
So with the current implementation you will need to:

1) Set FlyleafHost.VideoBackground with (1,0,0,0) solid color (A=1 from 255 to be able to use mouse events)
2) If you want to go fully transparent you can also force renderer's background with Config.Video.BackgroundColor and similary set A just 0 here

I didn't notice much performance penalties but probably there are so use this with caution! (E.g. I would use one player for transparency and one for normal and switch between those).

Let me know if you suggest a different way for this.

**Comment by LazyCuteLion** [comment]
Config.Video.BackgroundColor = new(0,0,0,0);//or other color with A=0
ClearRenderTargetView used it but will ignore the alpha and fill a solid color.(why???) 
so, can not "see" the background-image.
if for alpha-video,performance loss is acceptable.
i wish you can try: [1~4](https://github.com/SuRGeoNix/Flyleaf/issues/492#issue-2444414822) and [this](https://github.com/SuRGeoNix/Flyleaf/issues/492#issuecomment-2317058271)

**Comment by LazyCuteLion** [comment]
oh,add some info about webm(vp8/vp9).
need to set the codec to libvpx/libvpx-vp9 at the event of Player.Opening.
[check this](https://github.com/SuRGeoNix/Flyleaf/issues/637)
